reflex/vars.py

```diff
@@ -1393,8 +1394,15 @@ def contains(self, other: Any) -> Var:
                 raise VarTypeError(
                     f"'in <string>' requires string as left operand, not {other._var_type}"
                 )
+
+            _var_name = (
+                f"{self._var_name}.includes({other._var_full_name})"
+                if field is None
+                else f"{self._var_name}.some(e=>e.{field}==={other._var_full_name})"
```

Comment:
i think this should work if the field is specified as a state var, so maybe we can write it like

```
field = Var.create_safe(field, _var_is_string=isinstance(field, str))
```

and 

```python
f"{self._var_name}.some(e=>e[{field._var_full_name}]==={other._var_full_name})"
```

Reply:
I think the purpose is being able to write something like 
```python
rx.foreach(
    State.available_items,
    lambda item: rx.menu.item(
        rx.checkbox(
            item.name,
            checked=State.selected_items.contains(item.key, State.filter_key),
            on_change=lambda _: State.select_item(item),
        )
    ),
)
```
where you can change the value of `filter_key` dynamically

Reply:
Thanks @masenf, I will use your corrected version, it works with both strings and vars.

One thing that is that `e.{field}` could also take nested values like `field="address.street"` which will not work with the `e[...]` syntax.

I can also support this case, when I use this abomination:
```python
_var_name = f"{self._var_name}.some(e=>{field._var_name_unwrapped}.split(\".\").reduce((acc,v)=>acc[v],e)==={other._var_full_name})"

```

```python
class ComplexKey(rx.Base):
    part1: str
    part2: int

class ComplexItem(rx.Base):
    name: str
    color: str
    key: ComplexKey
```
Then you could write:
```python
State.selected_items.contains(item.key.part1, "key.part1"),
```


But at some point it becomes ridiculously complex...

-------- 8< --------- 8< --------------

And ideas for sleepless nights :sleeping_bed: :desktop_computer: 
Please nobody take this seriously :smile: 
```python
import reflex.js as js
...
checked=State.selected_items.some(
    js.λ("e", 
        js.var("e")
            .subscript(State.filter_key)                        
            .equals(js.var(State.search_text)
        )
    )
),
...
assert js.λ("e", 
        js.var("e")
            .subscript(State.filter_key)                        
            .equals(State.search_text)
        )
    ).__str__() == "e=>e[state__state.filter_key]===state__state.search_text"

assert is_turing_complete(js)
```

Reply:
🏆